phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableWithViewsIT.java

```diff
@@ -1216,5 +1218,248 @@ public void testDroppingIndexedColDropsViewIndex() throws Exception {
+        // extends to DDL timestamp
+        String schemaName = SCHEMA1;
+        String dataTableName = "T_" + generateUniqueName();
+        String viewName = "V_" + generateUniqueName();
+        String dataTableFullName = SchemaUtil.getTableName(schemaName, dataTableName);
+        String viewFullName = SchemaUtil.getTableName(schemaName, viewName);
+
+        String tableDDL = generateDDL("CREATE TABLE IF NOT EXISTS " + dataTableFullName + " ("
+            + " %s ID char(1) NOT NULL,"
+            + " COL1 integer NOT NULL,"
+            + " COL2 bigint,"
+            + " CONSTRAINT NAME_PK PRIMARY KEY (%s ID, COL1)"
+            + " ) %s");
+
+        String viewDDL = "CREATE VIEW " + viewFullName + " AS SELECT * FROM " + dataTableFullName;
+
+        String divergeDDL = "ALTER VIEW " + viewFullName + " DROP COLUMN COL2";
+        String viewColumnAddDDL = "ALTER VIEW " + viewFullName + " ADD COL3 varchar(50) NULL ";
+        String viewColumnDropDDL = "ALTER VIEW " + viewFullName + " DROP COLUMN COL3 ";
+        String tableColumnAddDDL = "ALTER TABLE " + dataTableFullName + " ADD COL4 varchar" +
+            "(50) NULL";
+        String tableColumnDropDDL = "ALTER TABLE " + dataTableFullName + " DROP COLUMN COL4 ";
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute(tableDDL);
+            conn.createStatement().execute(viewDDL);
+            long viewDDLTimestamp = getLastDDLTimestamp(conn, viewFullName);
+            Thread.sleep(1);
+            conn.createStatement().execute(divergeDDL);
+            //verify DDL timestamp changed
+            viewDDLTimestamp = CreateTableIT.verifyLastDDLTimestamp(schemaName, viewName,
+                viewFullName, viewDDLTimestamp, conn);
+            conn.createStatement().execute(viewColumnAddDDL);
+            //verify DDL timestamp changed because we added a column to the view
+            viewDDLTimestamp = CreateTableIT.verifyLastDDLTimestamp(schemaName, viewName,
+                viewFullName, viewDDLTimestamp, conn);
+            conn.createStatement().execute(viewColumnDropDDL);
+            //verify DDL timestamp changed because we dropped a column from the view
+            viewDDLTimestamp = CreateTableIT.verifyLastDDLTimestamp(schemaName, viewName,
+                viewFullName, viewDDLTimestamp, conn);
+            conn.createStatement().execute(tableColumnAddDDL);
+            //verify DDL timestamp DID NOT change because we added a column from the base table
+            assertEquals(viewDDLTimestamp, getLastDDLTimestamp(conn, viewFullName));
+            conn.createStatement().execute(tableColumnDropDDL);
+            assertEquals(viewDDLTimestamp, getLastDDLTimestamp(conn, viewFullName));
+        }
+    }
+
+    @Test
+    public void testLastDDLTimestampWithChildViews() throws Exception {
+        Assume.assumeTrue(isMultiTenant);
```

Comment:
Any reason why we want this test to run only if multiTenant is true ? I am not that much aware of this testsuite.
If we want to ensure isMultiTenant is true then can we change the test name to testLastDDLTimestampWith_Tenant_ChildViews to be more specific.

Reply:
@gjacoby126  This is not a comment for changing anything. This is just for my knowledge.

Reply:
@shahrs87 - the test deals in part with tenant-owned views, which don't make sense on a non-MT table.